web/profiles/custom/yalesites_profile/modules/custom/ys_themes/ys_themes.module

```diff
@@ -158,3 +158,39 @@ function ys_themes_build_css_variables() : string {
   $css[] = '}';
   return implode(PHP_EOL, $css);
 }
+
+/**
+ * Implements hook_form_alter().
+ *
+ * Adds a placeholder to the search form and visually hides the submit button.
+ */
+function ys_themes_form_alter(&$form, &$form_state, $form_id) {
+  if ($form_id == "views_exposed_form" && $form['#id'] == "views-exposed-form-search-page-1") {
+    $form['keywords']['#attributes'] = ['placeholder' => [t("Search this site")]];
+    $form['actions']['submit']['#attributes']['class'][] = "visually-hidden";
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_form_element_label_alter().
+ *
+ * Add a template to visually hide the search form label and add custom icon.
+ */
+function ys_themes_theme_suggestions_form_element_label_alter(array &$suggestions, array $variables, $hook) {
+  if (isset($variables['element']['#id']) && strpos($variables['element']['#id'], "edit-keywords") === 0) {
+    $suggestions[] = $hook . '__search_form';
+  }
+}
+
+/**
+ * Implements hook_metatags_alter().
+ *
+ * If there is a search term, change the title to reflect what was searched for.
+ */
+function ys_themes_metatags_alter(array &$metatags, array &$context) {
+  if (\Drupal::service('path.current')->getPath() == "/search") {
+    if (isset($_GET['keywords']) && $_GET['keywords'] != '') {
+      $metatags['title'] = t("Search for {$_GET['keywords']} | [site:name]");
+    }
+  }
+}
```

Comment:
The Metatag module is enabled on this site manage these sort of settings. I think we should enable the "Metatag: Views" module so that we can set this pattern within Views. We want to embrace the low-code site building tools whenever possible on this project.

Also just a side note that I get spooked whenever I see $_GET global. It's rarely an issue, but this is one of those random things that leads to possible security issues. (Usually when writing to a DB or interacting with files.) So I tend to always write a validation function when using user-input value. And I typically get these values from symfony instead of using the $_GET directly.

Reply:
@nJim - I actually tried the Metatag Views module first, but unfortunately it didn't parse the twig replacement ({{ arguments.search_api_fulltext }}) for the title as we wanted. It left it alone. So, I decided to do it in code. If you think there's another way we can handle this, let me know. I will definitely look into replacing the $_GET, thanks for that suggestion.